sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -74,9 +75,18 @@ private[sql] class HiveSessionCatalog(
   // | Methods and fields for interacting with HiveMetastoreCatalog |
   // ----------------------------------------------------------------
 
-  override def getDefaultDBPath(db: String): String = {
-    val defaultPath = context.hiveconf.getVar(HiveConf.ConfVars.METASTOREWAREHOUSE)
-    new Path(new Path(defaultPath), db + ".db").toString
+  // This function is to get the path for creating a non-default database
+  override def getDatabasePath(dbName: String, path: Option[String] = None): String = {
+    val dbPath = path.map(new Path(_)).getOrElse {
+      val defaultPath = context.hiveconf.getVar(HiveConf.ConfVars.METASTOREWAREHOUSE)
+      if (StringUtils.isBlank(defaultPath)) {
+        throw new AnalysisException(
+          s"${HiveConf.ConfVars.METASTOREWAREHOUSE.varname} is not set in the config or blank")
+      }
+      new Path(new Path(defaultPath), dbName.toLowerCase() + ".db")
+    }
+    val fs = dbPath.getFileSystem(context.hiveconf)
+    new Path(fs.getUri.getScheme, fs.getUri.getAuthority, dbPath.toUri.getPath).toString
```

Comment:
`dbPath.toString` does not have the scheme. For example, below is an example to show the differences.
```
/private/var/folders/4b/sgmfldk15js406vk7lw5llzw0000gn/T/spark-7e58f70e-168e-4416-8de4-f6040ffc79e9
file:/private/var/folders/4b/sgmfldk15js406vk7lw5llzw0000gn/T/spark-7e58f70e-168e-4416-8de4-f6040ffc79e9
```